### DISCUSSION TITLE
英文模版标题如何更换为“Chapter One”

### DISCUSSION THREAD

**Discussion by YZ-SJTU** [sjtug/SJTUThesis#614] [提问] [answered]
你好！

今日发布的最新模板中，对于英文撰写的论文，要求标题从Chapter 1 更改为Chapter One，请问应该在cls/ltx文件中哪里进行修改！

谢谢

**Comment by skyzh** [comment]
参考 https://tex.stackexchange.com/questions/29568/how-to-change-the-chapter-number-to-display-as-text-one-two

```
\usepackage{fmtcount}
\renewcommand{\thechapter}{\Numberstring{chapter}}
```

当然这会导致后面的章节编号变成 One.1.1 的形式。是否能提供一个最新模版的链接供参考？

**Reply by YZ-SJTU** [reply]
对的，我也发现了这个问题。https://bysj.jwc.sjtu.edu.cn/shownews.aspx?newsno=PdDCPqnlx72jC/3PDUitGA....

**Reply by skyzh** [reply]
这个页面无法打开。看看能不能下载然后上传到 GitHub？

**Reply by YZ-SJTU** [reply]
需要把后面四个点一起输进去）

**Comment by AlexaraWu** [comment, marked as answer]
不要直接改 `\thechapter`，可以用 `ctex` 提供的接口：
```latex
\usepackage{fmtcount}
\ctexset{%
  chapter = {%
    number = \Numberstring{chapter},
  }
}
\pdfstringdefDisableCommands{%
  \let\Numberstring\arabic
}
```

新版这个修改感觉怪怪的，有机会最好能和教务处的老师讨论一下。

**Reply by YZ-SJTU** [reply]
有一个warning
![image](https://user-images.githubusercontent.com/52121613/118461173-fa8fb500-b72f-11eb-9d41-4f70874d1485.png)

**Reply by YZ-SJTU** [reply]
ok  我解决了！
`\texorpdfstring{\Numberstring{chapter}}{}`

**Reply by AlexaraWu** [reply]
这么改的话 PDF 书签里的序号就没了。我答案重新编辑了一下。

**Reply by YZ-SJTU** [reply]
ok.感谢！

**Reply by YZ-SJTU** [reply]
> 这么改的话 PDF 书签里的序号就没了。我答案重新编辑了一下。

但好像有新的报错，
![image](https://user-images.githubusercontent.com/52121613/118463165-0c725780-b732-11eb-8c64-5d40cb3d3953.png)

**Reply by AlexaraWu** [reply]
如果你不熟悉 `cls` 文件的话就不要去改它，把我给的代码放在 `sjtusetup.tex` 的最后面。
